File: src/dataset/javascript/robot-name_exercism-ThomasZumsteg/openAI/ChatGPT4_robot-name_v1.js
const Robot = function() {
    this.name = getName();
};

Robot.prototype.reset = function() {
    // Release the old name and assign a new one
    namesInUse.delete(this.name);
    this.name = getName();
};

// Keeps track of robot names in use
const namesInUse = new Set();
const availableNames = generateAllNames();

function generateAllNames() {
    const names = [];
    for (let i = 0; i < 26; i++) {
        for (let j = 0; j < 26; j++) {
            for (let k = 0; k < 1000; k++) {
                const name = String.fromCharCode(65 + i, 65 + j) + k.toString().padStart(3, '0');
                names.push(name);
            }
        }
    }
    return names;
}

function getName() {
    if (availableNames.length === 0) {
        throw new Error("No more unique names available");
    }
    const index = Math.floor(Math.random() * availableNames.length);
    const name = availableNames.splice(index, 1)[0];
    namesInUse.add(name);
    return name;
}

module.exports = Robot;

File: src/dataset/python/robot-name_exercism-fruit-in/openAI/ChatGPT4_robot_name_v1.py
from random import randint


class Robot:
    def __init__(self):
        self.name = ''
        self.old_names = set()
        self.reset()

    def reset(self):
        while True:
            self.name = f"{chr(randint(65, 90))}{chr(randint(65, 90))}{randint(0, 999):03}"
            if self.name not in self.old_names:
                self.old_names.add(self.name)
                break